fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/RecordQueryPlannerConfiguration.java

```diff
@@ -238,6 +241,24 @@ public boolean isRuleEnabled(@Nonnull CascadesRule<?> rule) {
         return !disabledTransformationRules.contains(rule.getClass());
     }
 
+    /**
+     * Return whether the query planner should defer cross products during join enumeration.
+     * <br>
+     * Cross products are joins of data sources that are completely independent of one another, i.e., the data sources
+     * are not <em>connected</em> via a join predicate, nor is one correlated to the other. It is usually better to
+     * plan a cross product as the last join in a join tree as a cross product tends to increase the cardinality.
+     * Thus, such a join does not need to be planned among all the other joins and can therefore avoid useless
+     * exploration and over-enumeration.
+     * <br>
+     * Unfortunately, there are cases, however, where it may be beneficial to treat a cross product like any other
+     * join in a query.
```

Comment:
Is there a one sentence example that would fit in this comment, rather than leaving it elusive?

I think it's something like two (or more) low cardinality dimension tables and some (presumably very selective) predicates on the fact table.